sample/structured.h

```diff
@@ -50,6 +72,36 @@ struct StructuredData{
       
       return index;
     }
+    std::array<unsigned int,3> location_to_index_spherical(const Point<3,double> &p){
+      std::array<unsigned int,3> index;
+      for(int i=0;i<3;++i){
+        unsigned int temp_index=std::floor((p(i)-min[i])/spacing[i]);
+        if(((1.0*temp_index*spacing[i]+min[i])+(1.0*(temp_index+1)*spacing[i]+min[i]))/2 <= p(i)){
+          ++temp_index;
+        }
+        index[i]=temp_index;
+      }
+      
+      return index;
+      
+    }
+    std::array<double,3>
+    cartesian_to_spherical_coordinates(const Point<3> &position)
```

Comment:
in general, can you remove unnecessary empty lines within functions but add them between functions?